samples/led-matrix-weather/Program.Drawings.cs

```diff
@@ -0,0 +1,222 @@
+                case "02d":
+                    return IconPartiallySunny(uv, time);
+                case "02n": // partially cloudy at night
+                    return IconMoon(uv, time);
+                case "03d": // clouds day
+                case "03n": // clouds night
+                case "04d": // overcast clouds
+                case "04n":
+                case "11d": // with thunder
+                case "11n":
+                case "13d": // with snow
+                case "13n":
+                case "50d": // no clue what the picture represents but kinda looks like a cloud
+                case "50n":
+                    return IconClouds(uv, time);
+                case "09d":
+                case "09n":
+                case "10d": // with a bit of the sun
+                case "10n":
+                    return IconRain(uv, time);
+                default:
+                {
+                    // Don't know the icon
+                    // We're in Seattle so let's assume rain
+                    return IconRain(uv, time);
+                }
+            }
+        }
+
+        static Vector3 IconSunny(Vector2 uv, float time)
+        {
+            const int NumberOfRays = 10;
+            const float OuterRadius = 0.45f;
+            const float InnerRadius = 0.25f;
+
+            const float RayLength = OuterRadius - InnerRadius;
+
+            // spiral and rotate everything a bit
+            uv = Rot(uv, uv.Length() * 2 * 2 * pi / 10f * (float)Math.Sin(2 * pi * time / 9.0f) - time / 2.0f);
+            float angle = (float)Math.Atan2(uv.Y, uv.X) / 2 / pi * NumberOfRays;
+
+            // for soft rays: (float)Math.Sin(2 * pi * angle);
+            float rayShape = 2 * Math.Abs(mod(angle, 1.0f) - 0.5f);
+            float rayLen = RayLength * rayShape;
+            float radius = InnerRadius + rayLen;
+            float radiusDist = uv.Length() - radius;
+            return mix(
+                new Vector3(1, 1, 0),
+                new Vector3(0, 0, 0),
+                smoothstep(0f, 0.02f, radiusDist));
+        }
+
+        static float Clouds(Vector2 uv, float time)
+        {
+            float r0 = 0.15f;
+            Vector2 p0 = new Vector2(-0.2f, 0.1f + 0.05f * (float)Math.Sin(2 * pi * time / 7f + 0.1f));
+            float c0 = (uv - p0).Length() - r0;
+
+            float r1 = 0.15f + 0.05f * (float)Math.Sin(2 * pi * time / 13f);
+            Vector2 p1 = new Vector2(0f, 0.08f * (float)Math.Sin(2 * pi * time / 5f + 0.7f));
+            float c1 = (uv - p1).Length() - r1;
+
+            float r2 = 0.15f;
+            Vector2 p2 = new Vector2(0.2f, 0.05f - 0.05f * (float)Math.Sin(2 * pi * time / 4f + 0.9f));
+            float c2 = (uv - p2).Length() - r2;
+
+            float clouds = Math.Min(Math.Min(c0, c1), c2);
+            return smoothstep(0f, 0.2f, clouds + 0.05f);
+        }
+
+        static readonly Vector3 s_cloudsColor = new Vector3(0.3f, 0.3f, 0.3f);

```

Comment:
>static read [](start = 8, length = 11)

private

I'll stop adding these comments, so I'll assume you'll fix them all. thanks.